Kitodo/src/main/java/org/kitodo/production/services/data/ImportService.java

```diff
@@ -1394,4 +1415,66 @@ public boolean isRecordIdentifierMetadataConfigured(RulesetManagementInterface r
     public Collection<RecordIdentifierMissingDetail> getDetailsOfRecordIdentifierMissingError() {
         return recordIdentifierMissingDetails;
     }
+
+    /**
+     * Create and return list of "SelectItem" objects for given list of "ProcessDTO"s. For each "ProcessDTO" object
+     * check whether current user can link to it as a parent process.
+     * - If a specific process belongs to a project that is not assigned to the current user, add a hint to message to
+     *   the corresponding "SelectItem"
+     * - If the user cannot link to a specific process, because the process belongs to a project which is not assigned
+     *   to him, and he also lacks the special permission to link to processes in unassigned projects, the corresponding
+     *   "SelectItem" is disabled.
+     * @param parentCandidates list of "ProcessDTO"s
+     * @param maxNumber limit
+     * @return list of "SelectItem" objects corresponding to given "ProcessDTO" objects.
+     * @throws DAOException when checking whether user can link to given "ProcessDTO"s fails
+     * @throws IOException when opening ruleset file fails
+     */
+    public ArrayList<SelectItem> getPotentialParentProcesses(List<ProcessDTO> parentCandidates, int maxNumber)
+            throws DAOException, IOException {
+        ArrayList<SelectItem> possibleParentProcesses = new ArrayList<>();
+        for (ProcessDTO process : parentCandidates.subList(0, Math.min(parentCandidates.size(), maxNumber))) {
```

Comment:
If we limit to 10 possible results here, it might happen, that we already filter out all possible parents (because they are not part of the first 10 hits) here and end up with an empty result list.The check for possible parents has to be apllied on all search results unfortunately.